src/public/javascripts/batteries.js

```diff
@@ -1,225 +1,225 @@
-  }
-});
+/**
+ * Batteries Modules
+ * @module Batteries
+ */
+const client = require('./public/javascripts/communication').recievedEmitter;
+
+/** @constant {Number} - Width of battery table */
+const TABLE_WIDTH = 8;
+
+/** @constant {Number} - Height of battery table */
+const TABLE_HEIGHT = 9;
+
+/** @constant {Number} - Number of Cells to be in Battery Index */
+const NUM_CELLS = TABLE_WIDTH * TABLE_HEIGHT;
+
+/** @constant {Number} - Maximum nominal voltage */
+const MAX_VOLTAGE = 4.2;
+
+/** @constant {Number} - Minimum nominal voltage */
+const MIN_VOLTAGE = 3.1;
+
+/** @constant {HTMLElement} - HTML Element of Table */
+const TABLE = document.getElementById('batteryTable');
+
+let batteries = [];
+/**
+ * Represents a single battery cell
+ * @constructor
+ * @param {String} name - name of battery cell
+ * @param {Number} row - row it belongs to
+ * @param {Number} col - column it belongs to
+ */
+function Battery(name, row, col) {
+  this.name = name;
+  this.voltage = 0.0;
+  this.temperature = null;
+  this.row = row;
+  this.col = col;
+
+  this.setVoltage = function setVoltage(voltage) {
+    this.voltage = voltage;
+  };
+
+  this.setTemp = function setTemp(temp) {
+    this.temperature = temp;
+  };
+
+  this.getVoltage = function getVoltage() {
+    return this.voltage.toFixed(4);
+  };
+
+  this.getTemp = function getTemp() {
+    return this.temperature;
+  };
+}
+/**
+ * Creates a row filled with battery cells
+ * @param {Number} rowNum - Row number to create
+ * @param {Number} numCells - Number of cells in row to create
+ */
+function createRow(rowNum, numCells) {
+  let row = document.createElement('tr');
+  for (let i = 0; i < numCells; i++) {
+    let cell = document.createElement('td');
+    cell.id = `${rowNum}${i}cell`;
+    row.appendChild(cell);
+  }
+  TABLE.appendChild(row);
+}
+
+/**
+ * Creates battery table
+ * @param {Number} numRows - Number of rows in table
+ * @param {*} numCells - Number of cells per row
+ */
+function createTable(numRows, numCells) {
+  for (let i = 0; i < numRows; i++) {
+    createRow(i, numCells);
+  }
+}
+
+/**
+ * Fills cell with battery cell data
+ * @param {Number} batteryIndex - Index of battery to fill
+ * @param {Number} row - Row of cell to fill
+ * @param {Number} col - Column of cell to fill
+ */
+function fillCell(batteryIndex, row, col) {
+  cell = document.getElementById(`${row}${col}cell`);
+  let list = document.createElement('ul');
+  let header = document.createElement('li');
+  header.innerHTML = `<b>${batteries[batteryIndex].name}</b>`;
+  list.appendChild(header);
+  let voltage = document.createElement('li');
+  voltage.id = `${row}${col}volt`;
+  voltage.innerHTML = `Voltage: ${batteries[batteryIndex].getVoltage()}`;
+  list.appendChild(voltage);
+  cell.appendChild(list);
+}
+
+/**
+ * Creates table of batteries with given dimentions
+ */
+function initBatteries() {
+  createTable(TABLE_HEIGHT, TABLE_WIDTH);
+  for (let i = 0; i < NUM_CELLS; i++) {

```

Comment:
```javascript
/* Wrote up a simpler way to do this entire function, I think you can get it down to about 5 lines.
You basically are displaying a 1d array in 2d, 
but instead of iterating through it in 1d, just flatted your 2d index so its a lot simpler. 
Avoids a ton of repeated code, makes it easier to change later, and is much clearer to read. */
for (let i = 0; i < TABLE_HEIGHT; i++) {
    for (let j = 0; j < TABLE_WIDTH; i++) {
           let index = i * TABLE_WIDTH + j;   // Calculates what your flattened index is
           batteries[index] = new Battery(`Cell ${index}`, i, j);   // Adds cell in your 1d table
           fillCell(index, i, j));   // Fills in the cell, and column math looks a lot nicer because access is already in 2d
    }
}
```

Reply:
Good Suggestion, I wrote this at about 4 AM in California and I have no idea how it got in this PR but I'm all for optimizing the code :)